libs/base/Data/Fin.idr

```diff
@@ -54,6 +55,13 @@ finToNatInjective FZ     (FS _) Refl impossible
 finToNatInjective (FS m) (FS n) prf  =
   cong FS $ finToNatInjective m n $ succInjective (finToNat m) (finToNat n) prf
 
+||| Convert a Fin to a bounded Nat
+public export
+finToBoundedNat : Fin n -> Subset Nat (`LT` n)
+finToBoundedNat FZ     = Element 0 (LTESucc LTEZero)
+finToBoundedNat (FS x) = let Element k prf = finToBoundedNat x
+                          in Element (S k) (LTESucc prf)
+
```

Comment:
Being rebased to the current `master`, the second case could be simplified to
```idris
finToBoundedNat (FS x) = bimap S LTESucc $ finToBoundedNat x
```

Reply:
Neat! However, I'll rebase this and undo the addition of `finToBoundedNat`. After changing `bitsToIndex` to the more efficient type suggested by gallais, `finToBoundedNat` is unrelated to this PR.